src/Sulu/Bundle/EventLogBundle/Infrastructure/Doctrine/Repository/EventRecordRepository.php

```diff
@@ -62,13 +63,14 @@ public function createForDomainEvent(DomainEvent $domainEvent): EventRecordInter
         return $eventRecord;
     }
 
-    public function add(EventRecordInterface $eventRecord): void
+    public function addAndCommit(EventRecordInterface $eventRecord): void
     {
-        $this->entityManager->persist($eventRecord);
-    }
+        $this->entityManager->getUnitOfWork()->computeChangeSet(
```

Comment:
maybe add a comment
```
        // use persister to insert only given entity instead of flushing all managed entities via the entity manager
        // this prevents flushing unrelated changes and allows to call this method in a postFlush event-listener
```